src/client/Microsoft.Identity.Client/ManagedIdentity/CredentialProbeManager.cs

```diff
@@ -0,0 +1,112 @@
+
+        public CredentialProbeManager(RequestContext requestContext)
+        {
+            _requestContext = requestContext ?? throw new ArgumentNullException(nameof(requestContext));
+            _logger = _requestContext.Logger;
+        }
+
+        public async Task<bool> ExecuteProbeAsync(CancellationToken cancellationToken = default)
+        {
+            _logger.Info("[Probe] Initiating probe to IMDS credential endpoint.");
+
+            var request = new ManagedIdentityRequest(HttpMethod.Post, new Uri($"{CredentialEndpoint}?cred-api-version=1.0"))
+            {
+                Content = ProbeBody
+            };
+
+            HttpContent httpContent = request.CreateHttpContent();
+
+            _logger.Info($"[Probe] Sending request to {CredentialEndpoint}");
+            _logger.Verbose(() => $"[Probe] Request Headers: {string.Join(", ", request.Headers)}");
+            _logger.Verbose(() => $"[Probe] Request Body: {ProbeBody}");
+
+            try
+            {
+                IRetryPolicyFactory retryPolicyFactory = _requestContext.ServiceBundle.Config.RetryPolicyFactory;
+                IRetryPolicy retryPolicy = retryPolicyFactory.GetRetryPolicy(RequestType.ManagedIdentityDefault);
+
+                HttpResponse response = await _requestContext.ServiceBundle.HttpManager.SendRequestAsync(
+                    request.ComputeUri(),
+                    request.Headers,
+                    httpContent,
+                    request.Method,
+                    _logger,
+                    doNotThrow: true,
+                    mtlsCertificate: null,
+                    validateServerCertificate: null,
+                    cancellationToken, 
+                    retryPolicy).ConfigureAwait(false);
+
+                LogResponseDetails(response);
+
+                return EvaluateProbeResponse(response);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error($"[Probe] Exception during probe: {ex.Message}");
+                _logger.Error($"[Probe] Stack Trace: {ex.StackTrace}");
+                return false;
+            }
+        }
+
+        private void LogResponseDetails(HttpResponse response)
+        {
+            if (response == null)
+            {
+                _logger.Error("[Probe] No response received from the server.");
+                return;
+            }
+
+            _logger.Info($"[Probe] Response Status Code: {response.StatusCode}");
+            _logger.Verbose(() => $"[Probe] Response Headers: {string.Join(", ", response.HeadersAsDictionary)}");
+
+            if (response.Body != null)
+            {
+                _logger.Verbose(() => $"[Probe] Response Body: {response.Body}");
+            }
+        }
+
+        private bool EvaluateProbeResponse(HttpResponse response)
+        {
+            if (response == null)
+            {
+                _logger.Error("[Probe] No response received from the server.");
```

Comment:
Do not use `Error` because it is part of the regular control flow. It confused app developers. It is a warning at best.